src/helpers/loadOptimize.ts

```diff
@@ -0,0 +1,23 @@
+import { receiveExperiments } from '../app/featureFlags/actions';
+import { Store } from '../app/types';
+import { assertDocument } from '../app/utils';
+
+export const getOptimizeContainerByEnv = (window: Window) =>
+  window.location.hostname === 'openstax.org' ? 'OPT-NFHSM4B' : 'OPT-W65B3CP';
+
+export default (window: Window, store: Store) => new Promise((resolve) => {
+    const containerId = getOptimizeContainerByEnv(window);
+
+    const script = assertDocument().createElement('script');
+    script.setAttribute('type', 'text/javascript');
+    script.setAttribute('src', `https://www.googleoptimize.com/optimize.js?id=${containerId}`);
+    script.onload = resolve;
+    assertDocument().head.appendChild(script);
+
+    window.gtag('event', 'optimize.callback', {
+      callback: (idx: string, id: string) => {
+        store.dispatch(receiveExperiments([id, idx]));
```

Comment:
@TomWoodward I'm not sure how to mock the callback so that this line is covered by the test

Reply:
As I understand calling `window.gtag(...)` should push all arguments to the `window.dataLayer` so you could create a test in which you call callback from the first item, like `window.dataLayer[0][2].callback(idx, id)` and expect that receiveExperiments was dispatched

Reply:
That worked once I defined the function in test/setup.ts 👍